libs/application/templates/new-primary-school/src/lib/dataSchema.ts

```diff
@@ -41,14 +41,19 @@ export const dataSchema = z.object({
         .optional(),
     })
     .refine(
-      ({ usePronounAndPreferredName, preferredName }) =>
-        usePronounAndPreferredName.includes(YES) ? !!preferredName : true,
+      ({ usePronounAndPreferredName, preferredName, pronouns }) =>
+        usePronounAndPreferredName.includes(YES)
+          ? !!preferredName ||
+            (usePronounAndPreferredName.includes(YES) &&
+              pronouns &&
+              pronouns.length > 0)
```

Comment:
Do we need to check if `usePronounAndPreferredName.includes(YES)` again? 🤔 
```suggestion
            (pronouns && pronouns.length > 0)
```